syntax = "proto3";
option cc_enable_arenas = true;

package Ydb.Ping;
option java_package = "com.yandex.ydb.ping";
option java_outer_classname = "PingProtos";

import "ydb/public/api/protos/ydb_issue_message.proto";
import "ydb/public/api/protos/ydb_status_codes.proto";

// just go to GRPC without the rest of YDB

message PlainGrpcRequest {
}

message PlainGrpcResponse {
    StatusIds.StatusCode status = 1;
}

// Go until GrpcProxy

message GrpcProxyRequest {
}

message GrpcProxyResponse {
    StatusIds.StatusCode status = 1;
    repeated Ydb.Issue.IssueMessage issues = 2;
}

// Ping KQP proxy without executing anything

message KqpProxyRequest {
}

message KqpProxyResponse {
    StatusIds.StatusCode status = 1;
    repeated Ydb.Issue.IssueMessage issues = 2;
}
